frontend/src/advising/Templates/TemplateInfoPage.tsx

```diff
@@ -67,9 +83,150 @@ interface DropdownProps {
 
 interface NameFieldProps {
   readonly name: string;
-  readonly setTemplateName: (name: string) => void;
+  readonly setName: (name: string) => void;
+  readonly label: string;
+  readonly error: boolean;
 }
 
+interface FolderContext {
+  readonly folders: IFolderData[];
+  readonly setSelectedFolderId: (selectedFolderId: number | null) => void;
+  readonly newFolderName: string;
+  readonly setNewFolderName: (newFolderName: string) => void;
+  readonly hasDuplicateFolderName: boolean;
+}
+
+const folderContext = createContext<Partial<FolderContext>>({});
```

Comment:
Context for sharing state from the `NewTemplatesPage` component with its children (so we don't have to keep passing props down over and over)